src/System.Management.Automation/engine/remoting/client/RunspaceRef.cs

```diff
@@ -94,7 +95,22 @@ private PSCommand ParsePsCommandUsingScriptBlock(string line, bool? useLocalScop
                 // to be parsed and evaluated on the remote session (not in the current local session).
                 RemoteRunspace remoteRunspace = _runspaceRef.Value as RemoteRunspace;
                 bool isConfiguredLoopback = remoteRunspace != null && remoteRunspace.IsConfiguredLoopBack;
-                bool isTrustedInput = !isConfiguredLoopback && (localRunspace.ExecutionContext.LanguageMode == PSLanguageMode.FullLanguage);
+
+                bool inFullLanguage = context.LanguageMode == PSLanguageMode.FullLanguage;
+                if (context.LanguageMode == PSLanguageMode.ConstrainedLanguage
+                    && SystemPolicy.GetSystemLockdownPolicy() == SystemEnforcementMode.Audit)
+                {
+                    // In audit mode, report but don't enforce.
+                    inFullLanguage = true;
+                    SystemPolicy.LogWDACAuditMessage(
+                        context: context,
+                        title: RemotingErrorIdStrings.WDACGetPowerShellLogTitle,
+                        message: RemotingErrorIdStrings.WDACGetPowerShellLogMessage,
+                        fqid: "GetPowerShellMayFail",
+                        dropIntoDebugger: true);
+                }
+
+                bool isTrustedInput = !isConfiguredLoopback && inFullLanguage;
```

Comment:
I'm not sure about this - we're changing behavior with audit mode being enabled. Could you add a comment about why this is the right thing to do? If we're in CLM and because we're auditing now it's full language?

Reply:
The Audit mode enables `ConstrainedLangaugeMode`, but it only logs the behavior difference resulted by this `ConstrainedLanguageMode` setting, without actually enforcing them. So, from the user's perspective, the script still runs in `FullLanguageMode`.

So here, if `context.LanguageMode == PSLanguageMode.ConstrainedLanguage` and we are in Audit, then we don't enforce the `ConstrainedLanguageMode`, but instead, assuming it's in `FullLanauge`. So, the user's script runs as if it's in the `FullLanguageMode`, but they receive events about what will/may not work in the real `ConstrainedLanguage`.

Reply:
> we're changing behavior with audit mode being enabled.

Put it in another way, when in `Audit` mode, the script should behavior as if they are running in `FullLanguage` mode. We are not changing that behavior, but just logging what will/may fail in the real `ConstrainedLanguage` mode.

Reply:
I didn't understand that audit mode implies constrained language mode - do you think a comment explaining this should be placed here (or above)?

Reply:
A description about this should be put in the documentation of this Audit feature. I will check the documentation.

Reply:
It's already in the documentation at https://learn.microsoft.com/en-us/powershell/scripting/learn/application-control?view=powershell-7.4#wdac-policy-auditing:
> In audit mode, PowerShell runs the untrusted scripts in **Constrained Language** mode without errors, but logs messages to the event log instead. The log messages describe what restrictions would apply if the policy was in Enforce mode.